locked-asset/token-unstake/src/fees_accumulation.rs

```diff
@@ -120,21 +107,21 @@ pub trait FeesAccumulationModule:
         }
 
         let fees_mapper = self.fees_from_penalty_unlocking();
-        if fees_mapper.is_empty() {
+        let total_fees = fees_mapper.get();
+        if total_fees == 0u64 {
             last_send_mapper.set(current_epoch);
 
             return;
         }
 
-        let fees_attributes = fees_mapper.get();
         let energy_factory_addr = self.energy_factory_address().get();
-        let fee_tokens: EsdtTokenPayment = self
-            .energy_factory_proxy(energy_factory_addr)
-            .create_merged_locked_token_for_fees(
-                fees_attributes.token_amount,
-                fees_attributes.attributes.unlock_epoch,
-            )
-            .execute_on_dest_context();
+
+        let fee_tokens: EsdtTokenPayment = self.lock_virtual(
+            self.get_locked_token_id(&energy_factory_addr, BASE_TOKEN_ID_STORAGE_KEY),
+            total_fees,
+            self.blockchain().get_sc_address(),
+            energy_factory_addr,
+        );
```

Comment:
Fees collector should keep this total amount, and lock_virtual at the start of each claim week, not the unstake contract.

The flow should be:
- user unlocks, X tokens penalty
- unstake SC deposits those tokens in the fees collector SC
- fees collector checks at deposit time, that these are locked_tokens. If yes, it burns them, and adds them to a total, just like the other fungible tokens
- when a user collects rewards, it gives all fungible tokens as normal, but for locked tokens, it then calls virtual_lock

Reply:
Updated.